pkg/destroy/azure/azure.go

```diff
@@ -469,7 +469,10 @@ func deleteResourceGroup(ctx context.Context, client resources.GroupsClient, log
 		err = delFuture.WaitForCompletionRef(ctx, client.Client)
 	}
 	if err != nil {
-		if isNotFoundError(err) {
+		// FIXME. This is a workaround while using Azure SDK
+		// versions v5x and older which don't seem to pass
+		// the correct error code. They send a 0 instead of a 404.
+		if strings.Contains(err.Error(), "ResourceGroupNotFound") {
```

Comment:
I think we should go ahead and check `isNotFoundError`, too. That will cover us when we inevitably update the azure sdk and forget to change the logic here.

This is the code for `isNotFoundError` that worked for me to cover both SDK versions.
```
func isNotFoundError(err error) bool {
	if err == nil {
		return false
	}

	var dErr autorest.DetailedError
	errors.As(err, &dErr)

	if dErr.StatusCode == 404 {
		return true
	}

	if dErr.StatusCode == 0 {
		serviceErr, ok := dErr.Original.(*azure2.ServiceError)
		if ok {
			if strings.HasSuffix(serviceErr.Code, "NotFound") {
				return true
			}
		}
	}

	return false
}
```